[Zhenquan] iP

Impala36 opened the pull request.

Impala36 committed:
Level 0. Greet

Impala36 committed:
Level 0. Greet

Impala36 committed:
Level 2. Add, List

Impala36 committed:
Level 3. Mark as Done

Impala36 committed:
Level 4. ToDos, Events, Deadlines

Impala36 committed:
Level 5. Handle Errors
A-AbstractClasses
A-Packages

HongJiancheng commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,131 @@
+        return tasks.get(index);
+    }
+
+    public void changeDone(String line, UI ui, TaskList tasks) {
```
changeDone sounds a little bit weird, what about setDone?

Impala36 replied:
Yes, you're right! Thanks for the suggestion!

HongJiancheng commented on src/main/java/duke/ui/UI.java:
```diff
@@ -0,0 +1,55 @@
+        System.out.println(farewell);
+    }
+
+    public void markedAsDone(int index, TaskList tasks) {
```
markedAsDone is quite similar with TaskList Class' markAsDone, should we change to printDoneNotification or printDoneResult?

Impala36 replied:
Indeed this causes ambiguity, including my other print-related methods. Thanks!

HongJiancheng reviewed (commented):
Awesome codes! Lots of useful exceptions handled. Your codes inspired me a lot!

There are three minor suggestions, haha.

HongJiancheng commented on src/main/java/duke/ui/UI.java:
```diff
@@ -0,0 +1,55 @@
+
+import duke.task.TaskList;
+
+public class UI {
```
Although UI is short form of user interface, but base on coding standard, class name should be in PascalCase. Maybe Ui is better?

Impala36 replied:
Indeed it should follow PascalCase. I will make the changes.

Impala36 committed:
A-CodingQuality
Revamped many methods to look more readable,
avoided deep nesting,
happy path prominent,
refactored methods,
and SLAP.

Impala36 committed:
Level 6. Delete

Impala36 committed:
Level 7. Save

Impala36 committed:
A-JavaDoc
Added JavaDoc comments and removed redundant Ui method redBorderlines().

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into A-JavaDoc

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/ui/Style.java
#	src/main/java/duke/ui/Ui.java

Impala36 committed:
Edited JavaDoc comments.

Impala36 committed:
A-MoreOOP
Extracted all commands into classes.

Impala36 committed:
Level 8. Date and Time
Added date and time to deadline and events, renamed taskDescription to line to avoid confusion with description (actual description of task).

Impala36 committed:
Level 9. Find

Impala36 committed:
Level 9. Find

Impala36 committed:
A-JUnit
Added JUnit tests, imported Gradle.

Impala36 committed:
A-UserGuide

Impala36 committed:
Merge pull request #1 from Impala36/branch-Level-9

Branch level 9

Impala36 committed:
Merge pull request #2 from Impala36/branch-A-JUnit

Branch-A-JUnit

Impala36 committed:
A-UserGuide
Some formats were not working.

Impala36 committed:
Merge pull request #3 from Impala36/branch-A-UserGuide

A-UserGuide

Impala36 committed:
Update README.md

Impala36 committed:
Update README.md

Impala36 committed:
A-Gradle
A-CheckStyle
Restructured code to 0 gradle checkstyle violations

Impala36 committed:
Level-10
Changed all commands and UI to return Strings for GUI

Impala36 committed:
Changed isDone tick and cross to D and X due to encoding issues

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Impala36 committed:
Added gradle.yml

Impala36 committed:
BCD-Extension
Added D-Notes
Replaced all /n to /r due to DialogBox ellipsis issue
Removed unused text color functions

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Impala36 committed:
Update UserGuide.md

Impala36 committed:
Move variable declarations closer to usage

Scope of variables are large and may cause issues in future.

Variables are declared just before their usage to reduce their scope and smoothen the flow of thought for the reader.

Let's declare the variables in the same scope and just before usage.

Also add and update README.md, DeveloperGuide.md,  UserGuide.md, DoneCommandTest.java and NoteCommandTest.java.

Impala36 committed:
Change permission of gradlew

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Impala36 committed:
Remove redundant text-ui-test files

Program shifted to GUI so text UI tests are no longer required.

Impala36 committed:
Change permission of gradlew

Impala36 committed:
Merge pull request #5 from Impala36/branch-Level-10

Branch level 10

Impala36 committed:
Restyle to a medieval Duke theme

Makes the program more interesting to use.

Also update the README.md, DeveloperGuide.md, UserGuide.md.

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

Impala36 committed:
Change /r to \r and improve NoteCommand

Carriage returns were not displayed properly, and user inputting note without text is properly caught.

Impala36 committed:
Merge pull request #6 from Impala36/branch-Level-10

Branch level 10

Impala36 committed:
Update README.md

Impala36 committed:
Update UserGuide.md

Impala36 committed:
Update README.md

Impala36 committed:
Rename DukeGUI.png to Ui.png

Impala36 committed:
Merge branch 'master' into branch-Level-10

Impala36 committed:
Merge pull request #7 from Impala36/branch-Level-10

Rename DukeGUI.png to Ui.png

Impala36 committed:
Remove images folder
Shifted Ui.png and NoteSequenceDiagram.png out of images folder

Impala36 committed:
Merge branch 'master' into branch-Level-10

Impala36 committed:
Merge pull request #8 from Impala36/branch-Level-10

Remove images folder
